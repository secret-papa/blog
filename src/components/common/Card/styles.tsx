import styled from "styled-components";

export const Root = styled.div`
  width: 320px;
  background-color: #ffffff;
  box-shadow: rgba(0, 0, 0, 0.04) 0px 4px 16px 0px;
  border-radius: 4px;
`;

export const Header = styled.div``;

export const CoverImageWrap = styled.div`
  height: 168px;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

export const Body = styled.div`
  padding: 1rem;
`;

export const Title = styled.h4`
  font-size: 1rem;
  font-weight: bold;
  margin-bottom: 0.25rem;
  word-break: break-word;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  color: var(--color-grey-700);
`;

export const Description = styled.p`
  display: -webkit-box;
  height: 3.9375rem;
  font-size: 0.875rem;
  line-height: 1.5;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--color-grey-600);
  margin-bottom: 1.5rem;
`;

export const Meta = styled.div`
  font-size: 0.75rem;
  color: var(--color-grey-500);
`;
