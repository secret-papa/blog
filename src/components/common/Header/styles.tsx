import styled from "styled-components";

export const Wrapper = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 64px;
`;

export const Header = styled.div`
  display: flex;
  width: 1024px;
`;
