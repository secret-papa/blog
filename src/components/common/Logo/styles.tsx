import styled from "styled-components";

// eslint-disable-next-line import/prefer-default-export
export const Text = styled.span`
  font-size: 1.4rem;
  color: #2e4f4f;
`;
