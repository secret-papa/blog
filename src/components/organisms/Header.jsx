import React from "react";
import { css } from "@emotion/react";

import { Logo } from "../atoms";

const makeStyles = () => ({
  header: css`
    align-items: center;
    display: flex;
    justify-content: left;
    height: 7.2rem;
  `,
  content: css`
    margin: 0 auto;
    width: 110rem;
  `,
  header__logo: css``,
});

const Header = () => {
  const styles = makeStyles();

  return (
    <header css={styles.header}>
      <div css={styles.content}>
        <div css={styles.header__logo}>
          <Logo />
        </div>
      </div>
    </header>
  );
};

export default Header;
