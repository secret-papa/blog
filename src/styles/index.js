export { default as resetStyle } from "./reset";
export { default as theme } from "./theme";
