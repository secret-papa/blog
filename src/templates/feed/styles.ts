import styled from "styled-components";

export const Root = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
  background-color: #f8f9fa;
`;

export const Body = styled.main`
  width: 1024px;
`;

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  gap: 2rem;
`;
