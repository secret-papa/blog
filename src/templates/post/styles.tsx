import styled, { css } from "styled-components";
import { Link } from "gatsby";

export const Root = styled.div`
  margin-bottom: 3rem;
`;

export const Title = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: var(--color-grey-900);
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 5.5rem;
`;

export const Article = styled.article`
  width: 768px;
`;

export const CoverImageWrapper = styled.div`
  margin-top: 2rem;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
`;

export const MarkdownWrapper = styled.div`
  margin: 5rem 0 6rem;
`;

export const MarkdownBody = styled.div`
  & ol,
  ul {
    list-style: decimal;
  }
`;

export const NavigationButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  gap: 1rem;
`;

const leftDirectionCss = css`
  text-align: left;
`;
const rightDirectionCss = css`
  text-align: right;
`;

type Align = "left" | "right";

export const NavigationButton = styled(Link)<{ align?: Align }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 0.5rem;
  width: 100%;
  height: 4rem;
  background-color: #f8f9fa;
  padding-left: 1rem;
  padding-right: 1rem;
  box-shadow: rgba(0, 0, 0, 0.06) 0px 0px 4px 0px;
  border: none;
  cursor: pointer;
  ${props => (props.align === "right" ? rightDirectionCss : leftDirectionCss)}
`;

export const NavigationButtonCaption = styled.span`
  font-size: 0.75rem;
  color: var(--color-grey-600);
`;

export const NavigationButtonTitle = styled.h3`
  font-size: 1.125rem;
  font-weight: bold;
  color: var(--color-grey-700);
`;
