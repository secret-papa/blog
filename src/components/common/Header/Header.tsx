import React from "react";
import { Link } from "gatsby";

import { Header as StyledHeader, Wrapper } from "./styles";
import Logo from "../Logo";

const Header = () => (
  <Wrapper>
    <StyledHeader>
      <Link to="/">
        <Logo />
      </Link>
    </StyledHeader>
  </Wrapper>
);

export default Header;
