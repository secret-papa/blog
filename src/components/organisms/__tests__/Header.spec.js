import React from "react";

import { renderWithTheme } from "@tests";
import { theme } from "@src/styles";
import Header from "../Header";

describe("<Header />", () => {
  it("should render correctly", () => {
    const { container } = renderWithTheme(<Header />, theme);
    expect(container).toMatchSnapshot();
  });
});
