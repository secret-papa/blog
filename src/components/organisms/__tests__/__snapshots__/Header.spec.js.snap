// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: left;
  -ms-flex-pack: left;
  -webkit-justify-content: left;
  justify-content: left;
  height: 7.2rem;
}

.emotion-1 {
  margin: 0 auto;
  width: 110rem;
}

.emotion-3 {
  color: #757575;
  font: bold 3.2rem/1 sans-serif;
  font-family: inherit;
}

<div>
  <header
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <h1
          class="emotion-3"
        >
          :D
        </h1>
      </div>
    </div>
  </header>
</div>
`;
